hrms/hr/doctype/leave_encashment/leave_encashment.py

```diff
@@ -158,6 +158,8 @@ def set_leave_balance(self):
 		self.leave_allocation = allocation.name
 
 	def set_encashment_amount(self):
+		if not self.is_new():
+            		return	
```

Comment:
My suggestions:

- Add a field called "**Actual Encashment Amount**" (read-only) and keep Encashment Amount = Actual Encashment Amount
- Keep Encashment Amount editable
- Add patch to set Actual Encashment Amount = Encashment Amount for the current records

Another approach is to move the Encashment Amount to the Salary Structure Assignment doctype so that we can set employee-specific encashment amount just like different base and variable.